src/Compilers/VisualBasic/Portable/Parser/ParseConditional.vb

```diff
@@ -460,11 +477,50 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return SyntaxFactory.ReferenceDirectiveTrivia(hashToken, referenceKeyword, file)
         End Function
 
-        Private Shared Function ParseBadDirective(hashToken As PunctuationSyntax) As BadDirectiveTriviaSyntax
+        Private Function ParseShebangDirective(hashToken As PunctuationSyntax) As ShebangDirectiveTriviaSyntax
+            Dim builder = PooledStringBuilder.GetInstance()
+            Debug.Assert(CurrentToken.Kind = SyntaxKind.ExclamationToken,
+                         NameOf(ParseShebangDirective) & " called with wrong token")
+            Dim exclamationToken = DirectCast(CurrentToken, PunctuationSyntax)
+            GetNextToken()
+
+            ' Eat tokens until the end of line.
+            While CurrentToken.Kind <> SyntaxKind.EndOfFileToken AndAlso
+                  CurrentToken.Kind <> SyntaxKind.StatementTerminatorToken
+                builder.Builder.Append(CurrentToken.ToFullString())
+                GetNextToken()
+            End While
+
+
+            exclamationToken = exclamationToken.AddTrailingTrivia(SyntaxFactory.CommentTrivia(builder.ToStringAndFree()))
```

Comment:
I don't think we want the Kind of the trailing trivia to be "CommentTrivia" (since this is being implemented as a directive, not a type of comment).  C# has a PreprocessingMessageTrivia.  I think we'll probably want to add something similar in VB.